packages/@aws-cdk/aws-certificatemanager/lib/dns_validated_certificate_handler/index.js

```diff
@@ -0,0 +1,196 @@
+'use strict'
+
+const aws = require('aws-sdk')
+const UuidEncoder = require('uuid-encoder')
+const request = require('request-promise-native')
+
+const sleep = function (ms) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+/**
+ * Requests a public certificate from AWS Certificate Manager, using DNS validation.
+ * The hosted zone ID must refer to a **public** Route53-managed DNS zone that is authoritative
+ * for the suffix of the certificate's Common Name (CN).  For example, if the CN is
+ * `*.example.com`, the hosted zone ID must point to a Route 53 zone authoritative
+ * for `example.com`.
+ *
+ * @param {string} requestId the CloudFormation request ID
+ * @param {string} domainName the Common Name (CN) field for the requested certificate
+ * @param {string} hostedZoneId the Route53 Hosted Zone ID
+ * @returns {string} Validated certificate ARN
+ */
+const requestCertificate = async function (requestId, domainName, hostedZoneId) {
+  const acm = new aws.ACM()
+  const route53 = new aws.Route53()
+  const encoder = new UuidEncoder('base36')
+
+  console.log(`Requesting certificate for ${domainName}`)
+
+  const reqCertResponse = await acm.requestCertificate({
+    DomainName: domainName,
+    IdempotencyToken: encoder.encode(requestId),
```

Comment:
If you used a different way to come up with the idempotency token, you could get rid of all dependencies and we wouldn't need a package.json or build step anymore.

Why not simply use requestId as-is?

Reply:
It's not possible.  (I tried.)  There are (undocumented) length and character limits that apply to the token.

Reply:
Then why not using built-in libraries?

```
const sha = require('crypto').createHash('sha256');
sha.update(requestId);
const token = sha.digest('hex').substr(0, 32);
```

Reply:
The current method simply takes the request ID and encodes it using a reversible function that efficiently packs the value into a short string that fits within the token's allowed character set and length requirements.  

If I used the hash method you prescribe, we'd lose fidelity since the hash isn't reversible.  Whether that's important or not is admittedly debatable.